include/engine/routing_algorithms/routing_base.hpp

```diff
@@ -524,7 +524,11 @@ template <class DataFacadeT, class Derived> class BasicRoutingInterface
                                           std::vector<NodeID> &packed_path) const
     {
         NodeID current_node_id = middle_node_id;
-        while (current_node_id != search_heap.GetData(current_node_id).parent)
+        // all initial nodes will have itself as parent
+        // all incomplete paths (e.g. paths ending at the core boundary) will have SPECIAL_NODEID
+        // as parent
```

Comment:
That is a good point. In the current setup we always insert the start/target nodes in the non-core search heap, with themselves as parent. When we would relax that node, we check if its a core node and add it to the list of entry points, which are then added to the core search heap with `SPECIAL_NODEID` as parent. As such these nodes would never trigger a this code path.

We need to preserve these, thanks for finding this edge case. :+1: